workspaces/spectacle-shared/src/local-cli/client.ts

```diff
@@ -45,6 +55,10 @@ export class LocalCliSpectacle implements IForkableSpectacle {
     // send query to local cli-server
     return JsonHttpClient.postJson(`${this.baseUrl}/spectacle`, options);
   }
+
+  registerSpectacleUpdateEvent<T extends (...args: any) => any>(fn: T) {
+    this.spectacleUpdateEvents.push(fn);
+  }
```

Comment:
Probably best to pull in node's `EventEmitter` here if you're going the events + listeners way (with AsyncGenerators as the alternative) rather than building our own. It's already being imported and bundled in other places, so no reason to stay away from it for that. Also will allow your hook to deregister itself and not leak any handles.

Reply:
Interesting - I didn't realize that we could use the nodejs implementation - but apparently webpack includes this for us https://www.npmjs.com/package/events